android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java

```diff
@@ -978,6 +976,57 @@ public boolean doFireEvent(String event, Object data)
 		return krollObject.fireEvent(source, event, krollData, bubbles, reportSuccess, code, message);
 	}
 
+	/**
+	 * Our view proxy supports three properties to match iOS regarding
+	 * the text that is read aloud (or otherwise communicated) by the
+	 * assistive technology: accessibilityLabel, accessibilityHint
+	 * and accessibilityValue.
+	 *
+	 * We combine these to create the single Android property contentDescription.
+	 * (e.g., View.setContentDescription(...));
+	 */
+	public String composeContentDescription()
+	{
+		if (properties == null) {
+			return null;
+		}
+
+		final String punctuationPattern = "^.*\\p{Punct}\\s*$";
+		StringBuilder buffer = new StringBuilder();
+		String label = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_LABEL));
+		String hint = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_HINT));
+		String value = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_VALUE));
+
+		if (!TextUtils.isEmpty(label)) {
```

Comment:
These three small blocks are identical, extract to a method? Only wrinkle is the extra space prefix/suffix added if previous value was not empty...
```
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_LABEL));
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_HINT));
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_VALUE));
```